### PR TITLE
kraken: delete a queue after a timeout for a reload from kirin

### PR DESCRIPTION
**Pull request by kadhikari**
https://jira.kisio.org/browse/NAVP-969

### CODE REVIEW

File: source/kraken/maintenance_worker.cpp [hunk 1]
```diff
@@ -112,6 +112,8 @@ void MaintenanceWorker::load_realtime(){
         data_manager.get_data()->is_realtime_loaded = true;
     }else{
         LOG4CPLUS_WARN(logger, "no realtime data receive before timeout: going without it!");
+        // Incase of timeout we can delete the existing queue as we create a new queque in each call to the function
+        channel->DeleteQueue(queue_name);
```
Comment (kinnou02) [inline source/kraken/maintenance_worker.cpp]: You should do it every time, even when we got a response.

**Review by kinnou02 — CHANGES_REQUESTED** [review]

**Review by kinnou02 — APPROVED** [review]

File: source/kraken/maintenance_worker.cpp [hunk 2]
```diff
@@ -113,6 +113,8 @@ void MaintenanceWorker::load_realtime(){
     }else{
         LOG4CPLUS_WARN(logger, "no realtime data receive before timeout: going without it!");
     }
+    // Finally we delete de queue as we create a new queque in each call to the function
```
Comment (pbougue) [inline source/kraken/maintenance_worker.cpp]: typo: "Finally (maybe remove `we`) delete `th`e queue [...] new queue (no extra `q`) `at` each call"

**Review by pbougue — APPROVED** [review]
This PR is fine.
As for the JIRA ticket itself, how about other queues created by Kraken? Are there some that should be deleted also?
This can be a new ticket, but it would be nice to also check that.

File: source/kraken/maintenance_worker.cpp (hunk 2 above)
Comment (kadhikari) [reply source/kraken/maintenance_worker.cpp]: Done

**Pull request merged**